portal/src/PasswordField.tsx

```diff
@@ -0,0 +1,242 @@
+    });
+  }
+  if (passwordPolicy.minimum_guessable_level != null) {
+    policyData.push({
+      key: "minimum_guessable_level",
+      messageId: "PasswordField.minimum-guessable-level",
+      messageValues: {
+        level: passwordPolicy.minimum_guessable_level,
+        levelName:
+          guessableLevelNames[
+            passwordPolicy.minimum_guessable_level as GuessableLevel
+          ],
+      },
+    });
+  }
+  if (passwordPolicy.excluded_keywords != null) {
+    policyData.push({
+      key: "excluded_keywords",
+      messageId: "PasswordField.excluded-keywords",
+    });
+  }
+  if (passwordPolicy.history_size != null) {
+    policyData.push({
+      key: "history_size",
+      messageId: "PasswordField.history-size",
+      messageValues: { size: passwordPolicy.history_size },
+    });
+  }
+  if (passwordPolicy.history_days != null) {
+    policyData.push({
+      key: "history_days",
+      messageId: "PasswordField.history-days",
+      messageValues: { days: passwordPolicy.history_days },
+    });
+  }
+  return policyData;
+}
+
+export function extractGuessableLevel(
+  result: zxcvbn.ZXCVBNResult | null
+): GuessableLevel {
+  if (result == null) {
+    return 0;
+  }
+  return Math.floor(
+    Math.min(5, Math.max(1, result.score + 1))
+  ) as GuessableLevel;
+}
+
+const PasswordField: React.FC<PasswordFieldProps> = function PasswordField(
+  props: PasswordFieldProps
+) {
+  const {
+    className,
+    textFieldClassName,
+    value: password,
+    passwordPolicy,
+    ...rest
+  } = props;
+  const { renderToString } = useContext(Context);
+
+  const guessableLevelNames = useMemo(
+    () => renderGuessableLevelNames(renderToString),
+    [renderToString]
+  );
+  const passwordPolicyData = useMemo(
+    () => makePasswordPolicyData(passwordPolicy, guessableLevelNames),
+    [guessableLevelNames, passwordPolicy]
+  );
+
+  const result = useMemo(() => {
+    if (password != null && password !== "") {
+      return zxcvbn(password, passwordPolicy.excluded_keywords);
+    }
+    return null;
+  }, [password, passwordPolicy]);
+  const guessableLevel = extractGuessableLevel(result);
+
+  const isPasswordPolicySatisfied = useMemo(
+    () => checkPasswordPolicy(passwordPolicy, password, guessableLevel),
+    [password, passwordPolicy, guessableLevel]
+  );
+  return (
+    <div className={className}>
+      <TextField
+        {...rest}
+        value={password}
+        className={textFieldClassName}
+        type="password"
+      />
+      <PasswordStrengthMeter
+        level={guessableLevel}
+        guessableLevelNames={guessableLevelNames}
+      />
+      <ul className={styles.passwordPolicy}>
+        {passwordPolicyData.map((policy) => (
+          <li
+            key={policy.messageId}
+            className={cn(styles.policyLine, {
+              [styles.policySatisfied]: isPasswordPolicySatisfied[policy.key],
+            })}
+          >
+            <Text>
+              {"• "}
```

Comment:
Please use CSS to display the bullet. The bullet icon should not be included as part of content: it is a pure styling/display concern.

Reply:
Thanks, didn't notice that we can add bullet point by CSS